subsys/bluetooth/host/adv.c

```diff
@@ -1291,10 +1291,12 @@ int bt_le_adv_start_ext(struct bt_le_ext_adv *adv,
 	}
 
 	if (!dir_adv) {
+#if defined(CONFIG_BT_EXT_ADV)
 		err = bt_le_ext_adv_set_data(adv, ad, ad_len, sd, sd_len);
```

Comment:
Isn't it possible to use `if (IS_ENABLED(...)) {` instead of #ifdef?

Reply:
Yes, it should be, and would also be the right solution IMO

Reply:
I'll take a look, but what's the benefit of `if (IS_ENABLED())` vs `#if def` here?

Reply:
The benefit is that code inside `IS_ENABLED` are still compiled (thus catching compile bugs introduce if built without it being enabled) but is remove during linking. See https://docs.zephyrproject.org/latest/kernel/util/index.html#c.IS_ENABLED